src/core/layout/qgslayoutaligner.cpp

```diff
@@ -114,15 +175,16 @@ void QgsLayoutAligner::distributeItems( QgsLayout *layout, const QList<QgsLayout
         return itemBBox.center().y();
       case DistributeBottom:
         return itemBBox.bottom();
+      case DistributeHSpace:
+      case DistributeVSpace:
+        // not reachable branch, just to avoid compilation warning
+        return nan( "" );
```

Comment:
```suggestion
        return std::numeric_limits<double>::quiet_NaN();
```

Reply:
It was Matthias's suggestion :-D , I can change it again !

Reply:
Nan is fine, it's just nicer to use the more modern quiet_NaN approach :)